src/main/java/com/rultor/agents/github/CommentsTag.java

```diff
@@ -94,44 +106,191 @@ public CommentsTag(final Github ghub) {
+     * @param issue The Issue to add the comment to.
+     * @param tag The tag which is invalid.
+     * @throws IOException If there is a problem.
+     */
+    private void logInvalidTag(final Issue issue, final String tag)
+        throws IOException {
+        issue.comments().post(
+            String.format(
+                CommentsTag.PHRASES.getString("CommentsTag.invalid-tag"), tag
+            )
+        );
+        Logger.info(this, "tag %s is invalid", tag);
+    }
+
+    /**
+     * Logs a comment because the tag has already been released.
+     * @param issue The Issue to add the comment to.
+     * @param tag The tag that has already been released.
+     * @param rels The other releases.
+     * @param home The URI of home.
+     * @throws IOException If there is a problem.
+     * @checkstyle ParameterNumberCheck (15 lines)
+     */
+    private void logDuplicateTag(final Issue.Smart issue, final String tag,
+        final Releases.Smart rels, final URI home) throws IOException {
+        final Release.Smart rel = new Release.Smart(rels.find(tag));
+        rel.body(
+            String.format(
+                "%s\n\nSee also #%d and [build log](%s)",
+                rel.body(), issue.number(), home
+            )
+        );
+        issue.comments().post(
+            String.format(
+                CommentsTag.PHRASES.getString("CommentsTag.duplicate"),
+                tag
+            )
+        );
+        Logger.info(this, "duplicate tag %s commented", tag);
+    }
+
+    /**
+     * Valid version numbers:
+     * .1
+     * 2.2
+     * .1.2
+     * 1.2.3.4.5.6.7
+     *
+     * Invalid version numbers:
+     * abc
+     * a.b.c
+     * 1.
+     * 1.2.
+     * @param version Version number from a release
+     * @return True if the version is valid, false otherwise
+     */
+    private static boolean valid(final CharSequence version) {
+        return CommentsTag.VERSION_PATTERN.matcher(version).matches();
+    }
+
+    /**
+     * Is this tagged release valid?  A tagged release is valid if it's greater
+     * than any previous release.
+     * @param tag The release to be tagged
+     * @param previous The previous releases
+     * @return True if the release is valid
+     */
+    private static boolean valid(final String tag,
+        final Collection<DefaultArtifactVersion> previous) {
+        final DefaultArtifactVersion max;
+        if (previous.isEmpty()) {
+            max = new DefaultArtifactVersion("0");
+        } else {
+            max = Collections.max(previous);
+        }
+        return new DefaultArtifactVersion(tag).compareTo(max) == 1;
+    }
+
+    /**
+     * Transforms the Release into a DefaultArtifactVersion.
+     * @param release The Release to transform
+     * @return A DefaultArtifactVersion
+     * @throws IOException if there's a problem getting the release tag.
+     */
+    private static DefaultArtifactVersion version(final Release release)
```

Comment:
@gumbelmj don't use compound method names, see [here](http://www.yegor256.com/images/books/elegant-objects/seven-pages.pdf) for why.